ntbs-service/Models/TreatmentPeriod.cs

```diff
@@ -12,14 +12,16 @@ public class TreatmentPeriod
         public int? PeriodNumber { get; }
         public bool IsTransfer { get; }
         public List<TreatmentEvent> TreatmentEvents { get; }
-        public DateTime? PeriodStartDate { get; set; }
+        public DateTime? PeriodStartDate { get; }
         public DateTime? PeriodEndDate { get; set; }
```

Comment:
I think you can make the setter for `PeriodEndDate` private, since it should only be updated within the class